cmake/OpenCVFindLibsGUI.cmake

```diff
@@ -47,6 +47,13 @@ if(WITH_QT)
       find_package(Qt${QT_VERSION_MAJOR} COMPONENTS OpenGL QUIET)
       if(Qt${QT_VERSION_MAJOR}OpenGL_FOUND)
         set(QT_QTOPENGL_FOUND ON)  # HAVE_QT_OPENGL is defined below
+        if(QT_VERSION_MAJOR GREATER 5) # QGL -> QOpenGL
+          find_package(Qt${QT_VERSION_MAJOR} COMPONENTS OpenGLWidgets QUIET)
+          if(NOT Qt${QT_VERSION_MAJOR}OpenGLWidgets_FOUND)
+            message(STATUS "Qt OpenGLWidgets component not found: turning off openGL")
+            set(WITH_OPENGL FALSE) # Alternatively set(WITH_QT FALSE)
```

Comment:
> WITH_OPENGL

It is global option for all OpenCV modules.

`WITH_` variables should NOT be changed by detection scripts (its meaning - try to run auto-detection script for related feature).
We use `HAVE_` variables to specify that mentioned feature is available and enabled.

Perhaps we should run `set(QT_QTOPENGL_FOUND OFF)` here.
This variable is used in this script only to enable `HAVE_QT_OPENGL` below.

Reply:
Implemented change